packages/components/src/utils/getReferenceLinkHref.ts

```diff
@@ -22,41 +22,62 @@ const getSanitizedLinkHref = (url?: string) => {
 }
 
 // Convert the given reference link to the actual permalink
-const getActualLinkFromReference = (
-  referenceLink: string,
+const convertReferenceLinks = (
+  originalLink: string,
   sitemap: IsomerSitemap,
 ) => {
   const sitemapArray = getSitemapAsArray(sitemap)
-  const match = /\[resource:(\d+):(\d+)\]/.exec(referenceLink)
+  const match = /^\[resource:(\d+):(\d+)\]/.exec(originalLink)
 
   if (!match) {
-    return referenceLink
+    return originalLink
   }
 
   const refPageId = match[2]
 
   if (!refPageId) {
-    return referenceLink
+    return originalLink
   }
 
   const refPage = sitemapArray.find(({ id }) => id === refPageId)
 
   if (!refPage) {
-    return referenceLink
+    return originalLink
   }
 
   return refPage.permalink
 }
 
+// Prepend the assets base URL for asset links
+// Asset links are assumed to start with /{site_id}/
+const convertAssetLinks = (
+  originalLink: string,
+  assetsBaseUrl: string | undefined,
+) => {
+  if (!assetsBaseUrl) {
+    return originalLink
+  }
+
+  const match = /^\/(\d+)\//.exec(originalLink)
+
+  if (!match) {
+    return originalLink
+  }
+
+  return `${assetsBaseUrl}${originalLink}`
+}
+
 export const getReferenceLinkHref = (
   referenceLink: string | undefined,
   sitemap: IsomerSitemap,
+  assetsBaseUrl: string | undefined,
 ) => {
   if (!referenceLink) {
     return undefined
   }
 
-  const actualLink = getActualLinkFromReference(referenceLink, sitemap)
+  const assetLink = convertAssetLinks(referenceLink, assetsBaseUrl)
+  const actualLink = convertReferenceLinks(assetLink, sitemap)
```

Comment:
**no actionable required**. structure here feels abit off - i thought this would accept a `ReferenceLink` (or do some check to prove that it's a reference link) but the logic here is quite difficult to follow. 

i think we can fix this next time ba but we should probably have a top level function that accepts raw strings and spits out `reflink/extlink` etc as template types